### ISSUE TITLE
Extended tests are failing cmp tests

### ISSUE THREAD

**Issue by kroeckx** [openssl/openssl#12145]
In travis, the extended tests are failing with:
```
    # ERROR: (int) 'fixture->expected == ossl_cmp_msg_check_update(fixture->cmp_ctx, fixture->msg, NULL, 0)' failed @ test/cmp_vfy_test.c:127
    # [0] compared to [1]
    # check_msg_find_cert:crypto/cmp/cmp_vfy.c:538:CMP info: trying to verify msg signature with a valid cert that..
    # check_msg_find_cert:crypto/cmp/cmp_vfy.c:540:CMP info: matches msg sender    = /O=openssl_cmp
    # check_msg_find_cert:crypto/cmp/cmp_vfy.c:544:CMP info: while msg header does not contain senderKID
    # check_msg_all_certs:crypto/cmp/cmp_vfy.c:448:CMP info: trying first normal mode using trust store
    # check_msg_with_certs:crypto/cmp/cmp_vfy.c:401:CMP warning: no extraCerts
    # check_msg_with_certs:crypto/cmp/cmp_vfy.c:401:CMP warning: no untrusted certs
    # cert_acceptable:crypto/cmp/cmp_vfy.c:275:CMP info:  considering self-issued cert from certs in trusted store with..
    # cert_acceptable:crypto/cmp/cmp_vfy.c:277:CMP info:   subject = /O=openssl_cmp
    # cert_acceptable:crypto/cmp/cmp_vfy.c:296:CMP warning: cert has expired
    not ok 3 - test_validate_msg_signature_trusted_expired
```
[...]

```
65-test_cmp_vfy.t                (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
81-test_cmp_cli.t                (Wstat: 768 Tests: 7 Failed: 3)
  Failed tests:  4-5, 7
  Non-zero exit status: 3
```

@DDvO: Can you look at this?

**Comment by romen** [comment]
@mpeylo would you be able to look into this?

This comment might also be related: https://github.com/openssl/openssl/pull/11998#issuecomment-643693856

**Comment by DDvO** [comment]
I'm on vacation this week but will try to deal with this by tomorrow...

**Comment by DDvO** [comment]
How to reproduce these 'extended' Tests?

**Comment by DDvO** [comment]
BTW, I've recently added to `test/run_tests.pl` a `VERBOSE_FAILURES_ONLY` (short: `VFO`) env. var. option that allows to selectively show those sub-tests that failed. For instance.
```
 make test VFO=1 TESTS="test_cmp_*" 
```

**Comment by romen** [comment]
`VFO` looks really useful! Thanks for the heads up I did miss it!

To trigger the extended tests it is sufficient to have one of the commits in a `push` having

~~~
[extended tests] 
~~~

alone in a line of its message body.

What I often do to trigger them is adding an empty commit to be dropped at merge time just before pushing a new PR or every time I update a PR pushing some extra commits to address feedback:

~~~bash
; git commit --allow-empty -m 'drop! Trigger extended tests' -m '[extended tests]' --no-edit
~~~

**Comment by mouse07410** [comment]
@DDvO thanks for #12175, but I'm afraid it does not fix the issues I'm experiencing - number of errors and their kind remains the same after merging #12175:
```
Test Summary Report
-------------------
81-test_cmp_cli.t (Wstat: 1280 Tests: 7 Failed: 5)
  Failed tests:  3-7
  Non-zero exit status: 5
Files=1, Tests=7, 52 wallclock secs ( 0.18 usr  0.02 sys + 10.07 cusr  5.70 csys = 15.97 CPU)
Result: FAIL
```

Complete log: updated 
[81-out.txt](https://github.com/openssl/openssl/files/4792638/81-out.txt)

**Comment by DDvO** [comment]
> @DDvO thanks for #12175, but I'm afraid it does not fix the issues I'm experiencing - number of errors and their kind remains the same after merging #12175:

This was expected because #12175 is WIP which so far included a fix just for the issue with ubsan reported by @romen.

> Complete log: updated
> [81-out.txt](https://github.com/openssl/openssl/files/4792638/81-out.txt)

Good that you attached this output and that the output contains connection info like this:

**Comment by DDvO** [comment]
@kroeckx, even with the above hint by @romen on enabling extended tests I was not able to reproduce your issue at first.
Yet luckily even from the short test output section you quoted above I was able to infer what was going on: 

When fuzzing is enabled (`fuzz-libfuzzer` or `fuzz-afl` leading to `FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION` being defined) some CMP verification steps are omitted to allow for more aggressive fuzzing.
In this mode some tests make no sense, so I've disabled them.
Should be working now.

**Comment by mouse07410** [comment]
Darn...
> `...will contact http://127.0.0.1:1700/pkix/ via llproxy...`

I should've noticed that.

@DDvO thank you - I confirm your current #12175 works correctly. **Thanks!**

**Comment by DDvO** [comment]
@kroeckx, are the fixes in #12175 sufficient also for your issue here?

**Comment by DDvO** [comment]
Closing since fix is merged.
